lib/generators/boring/doorkeeper/devise/install/install_generator.rb

```diff
@@ -0,0 +1,182 @@
+            has_many :access_grants,
+                     class_name: 'Doorkeeper::AccessGrant',
+                     foreign_key: :resource_owner_id,
+                     dependent: :delete_all # or :destroy if you need callbacks
+
+            has_many :access_tokens,
+                     class_name: 'Doorkeeper::AccessToken',
+                     foreign_key: :resource_owner_id,
+                     dependent: :delete_all # or :destroy if you need callbacks
+          RUBY
+
+          inject_into_file "app/models/#{model_name}.rb",
+                           optimize_indentation(model_content, 2),
+                           after: "ApplicationRecord\n",
+                           verbose: false
+        end
+
+        def update_doorkeeper_initializer(model_name, grant_flows)
+          configure_resource_owner_authenticator(model_name) if grant_flows.include?("authorization_code")
+          configure_admin_authenticator(model_name) unless options[:api_only] || options[:skip_applications_routes]
+          configure_resource_owner_form_credentials(model_name) if grant_flows.include?("password")
+
+          gsub_file "config/initializers/doorkeeper.rb",
+                    /# grant_flows %w\[authorization_code client_credentials\]/,
+                    "grant_flows %w[#{grant_flows.join(' ')}]",
+                    verbose: false
+
+          if options[:api_only]
+            gsub_file "config/initializers/doorkeeper.rb",
+                      /# api_only/,
+                      "api_only",
+                      verbose: false
+          end
+
+          if options[:skip_applications_routes]
+            doorkeeper_routes_content = <<~RUBY
+              use_doorkeeper do
+                skip_controllers :applications, :authorized_applications
+              end
+            RUBY
+            
+            gsub_file "config/routes.rb",
+                      /.*use_doorkeeper/,
+                      optimize_indentation(doorkeeper_routes_content, 2),
+                      verbose: false
+          end
+
+          if options[:use_refresh_token]
+            uncomment_lines "config/initializers/doorkeeper.rb",
+                            /use_refresh_token/,
+                            verbose: false
+          end
+        end
+
+        def update_doorkeeper_migration(model_name, grant_flows)
+          uncomment_lines Dir["db/migrate/*_create_doorkeeper_tables.rb"].first,
+                          /add_foreign_key :oauth/,
+                          verbose: false
+          gsub_file Dir["db/migrate/*_create_doorkeeper_tables.rb"].first,
+                    /<model>/,
+                    ":#{model_name.pluralize}",
+                    verbose: false
+          return unless grant_flows.include?("password") || grant_flows.include?("client_credentials")
+
+          gsub_file Dir["db/migrate/*_create_doorkeeper_tables.rb"].first,
+                   /t.text    :redirect_uri, null: false/,
```

Comment:
Shouldn't this only happen for api only applications? Maybe we need to add this inside conditional as well?

Reply:
On another thought it also makes sense for it to be null since full stack app can also be used most of the time for serving APIs, will be a good idea to research more on this.

Reply:
redirect_uri is only required on the authorization code grant flow